plugin/trino-snowflake/src/test/java/io/trino/plugin/snowflake/TestSnowflakeConnectorTest.java

```diff
@@ -0,0 +1,343 @@
+            }
+        };
+    }
+
+    private void execute(String sql)
+            throws SQLException
+    {
+        try (Connection connection = DriverManager.getConnection(snowflakeServer.getJdbcUrl(), snowflakeServer.getProperties());
+                Statement statement = connection.createStatement()) {
+            statement.execute(sql);
+        }
+    }
+
+    @Override
+    protected boolean hasBehavior(TestingConnectorBehavior connectorBehavior)
+    {
+        switch (connectorBehavior) {
+            case SUPPORTS_AGGREGATION_PUSHDOWN:
+            case SUPPORTS_TOPN_PUSHDOWN:
+            case SUPPORTS_LIMIT_PUSHDOWN:
+                return false;
+            case SUPPORTS_COMMENT_ON_TABLE:
+                return false;
+            case SUPPORTS_ARRAY:
+                return false;
+            default:
+                return super.hasBehavior(connectorBehavior);
+        }
+    }
+
+    @Override
+    protected TestTable createTableWithDefaultColumns()
+    {
+        return new TestTable(
+                onRemoteDatabase(),
+                "tpch.table",
+                "(col_required BIGINT NOT NULL," +
+                        "col_nullable BIGINT," +
+                        "col_default BIGINT DEFAULT 43," +
+                        "col_nonnull_default BIGINT NOT NULL DEFAULT 42," +
+                        "col_required2 BIGINT NOT NULL)");
+    }
+
+    @Override
+    protected TestTable createTableWithUnsupportedColumn()
+    {
+        return new TestTable(
+                onRemoteDatabase(),
+                "tpch.test_unsupported_column_present",
+                "(one bigint, two decimal(50,0), three varchar(10))");
+    }
+
+    @Override
+    protected Optional<DataMappingTestSetup> filterDataMappingSmokeTestData(DataMappingTestSetup dataMappingTestSetup)
+    {
+        String typeName = dataMappingTestSetup.getTrinoTypeName();
+        // TODO: Test fails with type real
+        if (typeName.equals("real")) {
+            return Optional.empty();
+        }
+        // TODO: Test fails with DATE '1582-10-05'
+        if (typeName.equals("date")) {
+            if (dataMappingTestSetup.getSampleValueLiteral().equals("DATE '1582-10-05'")) {
+                return Optional.empty();
+            }
+        }
+        if (typeName.equals("time")
+                || typeName.equals("timestamp")
+                || typeName.equals("timestamp(3) with time zone")) {
+            return Optional.of(dataMappingTestSetup.asUnsupported());
+        }
+        return Optional.of(dataMappingTestSetup);
+    }
+
+    @Override
+    protected boolean isColumnNameRejected(Exception exception, String columnName, boolean delimited)
+    {
+        return nullToEmpty(exception.getMessage()).matches(".*(Incorrect column name).*");
+    }
+
+    @Test
+    @Override
+    public void testShowColumns()
+    {
+        MaterializedResult actual = computeActual("SHOW COLUMNS FROM orders");
```

Comment:
Please leave comment why overriding. Subsequent tests are same.

Reply:
Added comment for this override.

Reply:
Reminder "Subsequent tests are same."

Reply:
Added comments for all the other override tests.